### PR TITLE
Introduce a first set of blockchain apis

### PR DESCRIPTION
**Pull request by chb0github**
## PR Checklist

- [x] I've built the code `cargo build`. For major changes, `cargo build --workspace --release` is recommended.
- [x] I've run the automated unit tests `cargo test`. This executes our automated unit tests.
- [x] I've not broken any existing tests or functionality. In addition to `cargo test`, I've run `cargo clippy`
- [x] I've not introduced any new known security vulnerabilities. I've run `cargo audit`

- [x] I've included a brief description and a good title of the proposed changes and how to test them.
- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've associated an [issue](https://github.com/pyrsia/pyrsia/issues) with this Pull Request. 
- [x] I've checked that the code is up-to-date with the `pyrsia/main` branch.
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've assigned this Pull Request to "pyrsia/collaborators"

## Description

Fixes pyrsia/pyrsia#321

This PR represents an initial stab at a set of blockchain APIs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ target/
 
 .vscode/
 .DS_Store
+/src/blockchain/blockchain_storage
```
Comment (prince-chrismc) [inline .gitignore:18]: This should not be committed, the test code should be fixed.

We should open a separate issue for that

**Review by prince-chrismc — COMMENTED** [review]
Just the one nit, otherwise ✅

**Review by fishseabowl — APPROVED** [review]
Good work!

File: .gitignore (hunk 1 above)
Comment (chb0github) [reply .gitignore:18]: test code should manage the block entirely in memory

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**